esp-hal/src/uart.rs

```diff
@@ -904,6 +904,24 @@ where
         Ok(to_read)
     }
 
+    /// Busy waits for a break condition to be detected on the RX
+    /// line. Condition is met when the receiver detects a NULL character
+    /// (i.e. logic 0 for one NULL character transmission) after stop bits.
+    ///
+    /// Clears the break detection interrupt before returning.
+    #[instability::unstable]
+    pub fn wait_for_break(&mut self) {
```

Comment:
- Why do you need the `int_ena` bit to be set?
- You shouldn't clear `int_ena` at the end of the function without actually checking if it was cleared before
- This function can end up hanging if the user has an interrupt handled that clears the brk_det bit. I know it would be their own fault but we should try our best to minimize the change of bugs.

Reply:
- Why do you need the int_ena bit to be set?
  - I am operating under the assumption that the `int_raw` only functions if `int_ena` is set. Is this incorrect?
- You shouldn't clear int_ena at the end of the function without actually checking if it was cleared before
  - Sorry, do you mean `int_clr`?
- This function can end up hanging if the user has an interrupt handled that clears the brk_det bit. I know it would be their own fault but we should try our best to minimize the change of bugs.
  - Makes sense

Reply:
> Is this incorrect?

No, as far as I know only `int_st` and the actual interrupt handling is affected, where `int_st = int_raw & int_ena`. Usually.

> Sorry, do you mean int_clr?

Hmm yeah disregard that point, I saw something else, not what you wrote. But in that case you should disable listening for the interrupt :)